librosa/core/spectrum.py

```diff
@@ -1168,8 +1168,17 @@ def magphase(D, *, power=1):
     """
 
     mag = np.abs(D)
+
+    # Prevent NaNs and return magnitude 0, phase 1+0j for zero
+    zeros_to_ones = mag == 0
+    mag_nonzero = mag + zeros_to_ones
+    # Compute real and imaginary separately, because complex division can
+    # produce NaNs when denormalized numbers are involved (< ~2e-39 for
+    # complex64, ~5e-309 for complex128)
+    phase = D.imag / mag_nonzero * 1.0j
+    phase += D.real / mag_nonzero + zeros_to_ones
```

Comment:
I wonder if this could be faster by working directly with the components of the phase array to avoid an extra multiply?

```python
phase = np.empty_like(D)
phase.imag = D.imag / mag_nonzero
phase.real = D.real / mag_nonzero + zeros_to_ones
```